server/src/grpc/mod.rs

```diff
@@ -116,3 +118,234 @@ impl From<ArrErr> for Status {
+        match field {
+            GrpcField::String(field) => field,
+            _ => format!("{:?}", field),
+        }
+    }
+}
+impl From<GrpcField> for Vec<i64> {
+    fn from(field: GrpcField) -> Self {
+        match field {
+            GrpcField::I64List(field) => field,
+            GrpcField::I64(field) => vec![field],
+            _ => vec![],
+        }
+    }
+}
+impl From<GrpcField> for i64 {
+    fn from(field: GrpcField) -> Self {
+        match field {
+            GrpcField::I64(field) => field,
+            _ => 0,
+        }
+    }
+}
+impl From<GrpcField> for f64 {
+    fn from(field: GrpcField) -> Self {
+        match field {
+            GrpcField::F64(field) => field,
+            _ => 0.0,
+        }
+    }
+}
+impl From<GrpcField> for i32 {
+    fn from(field: GrpcField) -> Self {
+        match field {
+            GrpcField::I32(field) => field,
+            _ => 0,
+        }
+    }
+}
+impl From<GrpcField> for i16 {
+    fn from(field: GrpcField) -> Self {
+        match field {
+            GrpcField::I16(field) => field,
+            _ => 0,
+        }
+    }
+}
+impl From<GrpcField> for bool {
+    fn from(field: GrpcField) -> Self {
+        match field {
+            GrpcField::Bool(field) => field,
+            _ => false,
+        }
+    }
+}
+impl From<GrpcField> for Timestamp {
+    fn from(field: GrpcField) -> Self {
+        match field {
+            GrpcField::Timestamp(field) => field,
+            _ => Timestamp::from(SystemTime::now()),
+        }
+    }
+}
+impl From<GrpcFieldOption> for Option<GrpcField> {
+    fn from(field: GrpcFieldOption) -> Self {
+        match field {
+            GrpcFieldOption::Bytes(field) => field.map(GrpcField::Bytes),
+            GrpcFieldOption::StringList(field) => field.map(GrpcField::StringList),
+            GrpcFieldOption::String(field) => field.map(GrpcField::String),
+            GrpcFieldOption::I64List(field) => field.map(GrpcField::I64List),
+            GrpcFieldOption::I64(field) => field.map(GrpcField::I64),
+            GrpcFieldOption::F64(field) => field.map(GrpcField::F64),
+            GrpcFieldOption::I32(field) => field.map(GrpcField::I32),
+            GrpcFieldOption::I16(field) => field.map(GrpcField::I16),
+            GrpcFieldOption::F32(field) => field.map(GrpcField::F32),
+            GrpcFieldOption::Bool(field) => field.map(GrpcField::Bool),
+            GrpcFieldOption::Timestamp(field) => field.map(GrpcField::Timestamp),
+            GrpcFieldOption::None => None,
+        }
+    }
+}
+
+/// Starts the grpc servers for this microservice using the provided configuration
+///
+/// # Example:
+/// ```
+/// use svc_storage::common::ArrErr;
+/// use svc_storage::config::Config;
+/// use svc_storage::grpc::grpc_server;
+/// async fn example() -> Result<(), ArrErr> {
+///     let config = Config::default();
+///     tokio::spawn(grpc_server(config)).await?;
+///     Ok(())
+/// }
+/// ```
+pub async fn grpc_server(config: Config) {
```

Comment:
I contemplated doing so and decided not to because I only have a grpc server and no clients. But since you're asking now, I decided to keep this as close to the template as possible as well.

I started to make the change 2 days ago, but it turned out to be a bit more work than I initially thought. The includes of the generated `rust` files from `proto` expect to be part of a specific path to keep the `rust` docs working, and I've moved the macros that generate the GrpcService implementations to the `grpc` module as well, which introduced some fun bugs.

I've got a working example again now, so all seems good!
Changes can be found in eb7a15365303e90ec212c4828ec07b0e1332a678